### PR TITLE
Add bench for tanimoto_distance

### PR DESCRIPTION
**Pull request by JJ-Pineda**
Also modify process_cpd() method to export `Fingerprint` type instead of bitvec. This is necessary because the tanimoto_distance() method takes `Fingerprint` as input (i.e. not a bitvec).

### CODE REVIEW

File: benches/tanimoto_bench.rs [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+fn bench_tanimoto_distance(b: &mut Bencher) {
+    let smiles1 = "[N]Cc1cncc2c(=O)c3cccc(CCC(=O)O)c3[nH]c12";
+    let (_proc_smiles1, fingerprint1, _descriptors1) = process_cpd(smiles1).unwrap();
+
+    let smiles2 = "CCc1cccc2c(=O)c3cncc(CN)c3[nH]c12";
+    let (_proc_smiles2, fingerprint2, _descriptors2) = process_cpd(smiles2).unwrap();
+
+
+    b.iter(|| fingerprint1.tanimoto_distance(&fingerprint2));
+}
```
Comment (xrl) [inline benches/tanimoto_bench.rs:19]: how about benching the fingerprint generation as well?

File: src/analysis/compound_processing.rs [hunk 2]
```diff
@@ -30,14 +30,13 @@ pub fn get_tautomers(romol: &ROMol) -> Vec<ROMol> {
     ts
 }
 
-#[allow(unreachable_code)]
-pub fn process_cpd(smi: &str) -> eyre::Result<(&str, BitVec<u8>, HashMap<String, f64>)> {
+pub fn process_cpd(smi: &str) -> eyre::Result<(&str, Fingerprint, HashMap<String, f64>)> {
```
Comment (xrl) [inline src/analysis/compound_processing.rs:33]: Good catch!

File: benches/tanimoto_bench.rs (hunk 1 above)
Comment (JJ-Pineda) [reply benches/tanimoto_bench.rs]: Done!

**Review by xrl — APPROVED** [review]
Looks good to me!

**Review by xrl — CHANGES_REQUESTED** [review]
Actually, can you change it so benchmarks are only run when merged to main?

**Comment by xrl** [comment]
Here's an example of how to run a workflow on one specific branch: https://github.com/assaydepot/k3-eks-service-accounts/blob/main/.github/workflows/apply.yml#L1-L3

**Review by xrl — APPROVED** [review]
Let's merge it and see if the benchmarks work!

**Pull request merged**